### PR TITLE
deprecate Jpeg2KImageFile.reduce property

### PR DESCRIPTION
**Pull request by homm**

### CODE REVIEW

File: src/PIL/Jpeg2KImagePlugin.py [hunk 1]
```diff
@@ -217,11 +214,24 @@ def load(self):
...
 
+    @property
+    def reduce(self):
+        return super().reduce
+
+    @reduce.setter
+    def reduce(self, value):
+        warnings.warn(
+            "The reduce property for Jpeg2KImageFile is deprecated "
+            "and will be removed in the next major version.",
```
Comment (hugovk) [inline src/PIL/Jpeg2KImagePlugin.py]:
```suggestion
            "and will be removed in a future major version.",
```

Docs, emphasis added:

> Deprecated features are only removed in major releases **after an appropriate period of deprecation has passed**.

https://pillow.readthedocs.io/en/stable/deprecations.html#removed-features

Python 3.5 EOL is due in September, so Pillow 8 may be out in October (assuming a major bump for an EOL version drop).

We've gone for around a year for an "appropriate period" (https://github.com/python-pillow/Pillow/issues/1087#issuecomment-121655443) so let's keep the comment less specific.

Comment (homm) [reply src/PIL/Jpeg2KImagePlugin.py]: Which version this will be deprecated in? 7.0.1 or 7.0?

Comment (hugovk) [reply src/PIL/Jpeg2KImagePlugin.py]: 7.1.0 is the next one due out, in April.

Comment (homm) [reply src/PIL/Jpeg2KImagePlugin.py]: ok, updated deprecations.rst

**Comment by homm** [comment]
Any updates?

**Comment by radarhere** [comment]
I'm not enthusiastic about this, since in the unlikely event that anyone was relying on `im.reduce` to keep it's value once set -

```python
from PIL import Image
im = Image.open("Tests/images/16bit.cropped.j2k")
im.reduce = 5
print(im.reduce)  # 5
```

This PR would break it in Pillow 7.1.0 without warning.

```python
from PIL import Image
im = Image.open("Tests/images/16bit.cropped.j2k")
im.reduce = 5
print(im.reduce)  # <bound method Image.reduce of <PIL.Jpeg2KImagePlugin.Jpeg2KImageFile image mode=I;16 size=64x64 at 0x108C167B8>>
```

Feel free to try and convince others, or create your own parallel PR in the main repo.

**Comment by radarhere** [comment]
Closing, in favour of https://github.com/python-pillow/Pillow/pull/4474

**Pull request closed without merging**